src/meta-srv/src/procedure/create_table.rs

```diff
@@ -99,17 +111,117 @@ impl CreateTableProcedure {
             return Ok(Status::Done);
         }
 
-        self.creator.data.state = CreateTableState::DatanodeCreateTable;
+        self.creator.data.state = if expr.engine == MITO2_ENGINE {
+            CreateTableState::DatanodeCreateRegions
+        } else {
+            CreateTableState::DatanodeCreateTable
+        };
 
         Ok(Status::executing(true))
     }
 
-    async fn on_create_metadata(&self) -> Result<Status> {
-        let _timer = common_telemetry::timer!(
-            crate::metrics::METRIC_META_CREATE_TABLE_PROCEDURE_CREATE_META
-        );
+    fn create_region_request_template(&self) -> PbCreateRegionRequest {
+        let create_table_expr = &self.creator.data.task.create_table;
+
+        let mut column_defs = Vec::with_capacity(create_table_expr.column_defs.len());
+        let mut primary_key = Vec::with_capacity(create_table_expr.primary_keys.len());
+        create_table_expr
+            .column_defs
+            .iter()
+            .enumerate()
+            .for_each(|(i, c)| {
+                let semantic_type = if create_table_expr.time_index == c.name {
+                    SemanticType::Timestamp
+                } else if create_table_expr.primary_keys.contains(&c.name) {
+                    primary_key.push(i as u32);
```

Comment:
Does this respect the column order of the primary key? e.g. We have `PRIMARY_KEY(b, a)`. but in `column_defs` we store `column a, column b`.

Reply:
We should respect the user-defined primary key orders. The following expression should have a different primary key.
`create table foo (a int, b int, primary key(a, b))`  -> (a, b)
`create table bar (a int, b int, primary key(b, a))` -> (b, a)
If we store the primary key like `a + b` in an ordered tree structure. For `(a, b)`, We  have indexed search capabilities on `(a), (a, b)`,  For `(b, a)` We have indexed search capabilities on `(b), (b, a)`.